lib/ansible/modules/files/copy.py

```diff
@@ -477,7 +477,7 @@ def main():
             backup=dict(type='bool', default=False),
             force=dict(type='bool', default=True, aliases=['thirsty']),
             validate=dict(type='str'),
-            directory_mode=dict(type='raw'),
+            directory_mode=dict(type='mode'),
```

Comment:
Why isn't the parameter `mode` also defined here in the argspec? It's listed in the docs but not here.

Reply:
`AnsibleModule()` has an argument `add_file_common_args`, which will add common parameters such as `mode` to the argspec. The parametes which are added are the ones in `FILE_COMMON_ARGUMENTS` defined in `module_utils/basic.py`, which includes [this change](https://github.com/ansible/ansible/pull/50035/files#diff-90085fdcec6ed8b273ba885eaee60328R235).